custom_components/kia_uvo/Vehicle.py

```diff
@@ -98,7 +98,7 @@ def set_geocoded_location(self, old_vehicle_location):
         new_lat = self.get_child_value("vehicleLocation.coord.lat")
         new_lon = self.get_child_value("vehicleLocation.coord.lon")
 
-        if (old_lat != new_lat or old_lon != new_lon) or old_geocode is None:
+        if ((old_lat != new_lat or old_lon != new_lon) or old_geocode is None) and (old_lon != None and old_lat != None):
```

Comment:
Why do we need this additional condition? Enabling geolocation could be done using options and just before enabling it, it could be none.

Reply:
Typo - this should be checking for new_lon, was crashing before because new was empty.